### PR TITLE
feat(keyboard) add Home and End keys mapping

### PR DESCRIPTION
**Pull request by ThomasDevoogdt**
### Description of the feature or fix

Add `libinput`, `XKB` and `SDL` Home and End keys mapping.

### Notes
- Update the [Documentation](https://github.com/lvgl/lvgl/tree/master/docs) if needed.
- Add [Examples](https://github.com/lvgl/lvgl/tree/master/examples) if relevant.
- Add [Tests](https://github.com/lvgl/lvgl/blob/master/tests/README.md) if applicable.
- If you added new options to `lv_conf_template.h` run [lv_conf_internal_gen.py](https://github.com/lvgl/lvgl/blob/master/scripts/lv_conf_internal_gen.py) and update [Kconfig](https://github.com/lvgl/lvgl/blob/master/Kconfig).
- Run `scripts/code-format.py` ([astyle](http://astyle.sourceforge.net/install.html) needs to be installed) and follow the [Code Conventions](https://docs.lvgl.io/master/CODING_STYLE.html)
- Mark the Pull request as [Draft](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/changing-the-stage-of-a-pull-request) while you are working on the first version, and mark is as _Ready_ when it's ready for review.
- When changes were requested, [re-request review](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/requesting-a-pull-request-review) to notify the maintainers.
- Help us to review this Pull Request! Anyone can [approve or request changes](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/approving-a-pull-request-with-required-reviews).

### CODE REVIEW

File: src/drivers/libinput/lv_libinput.c [hunk 1]
```diff
@@ -587,6 +587,9 @@ static void _read_keypad(lv_libinput_t * dsc, struct libinput_event * event)
                 case KEY_TAB:
                     evt->key_val = LV_KEY_NEXT;
                     break;
+                case KEY_HOME:
```
Comment (PGNetHun) [inline src/drivers/libinput/lv_libinput.c:590]: Please add `KEY_END` also

File: src/drivers/libinput/lv_xkb.c [hunk 2]
```diff
@@ -120,6 +120,10 @@ uint32_t lv_xkb_process_key(lv_xkb_t * dsc, uint32_t scancode, bool down)
         case XKB_KEY_ISO_Left_Tab: /* Sent on SHIFT + TAB */
             result = LV_KEY_PREV;
             break;
+        case XKB_KEY_Home:
+        case XKB_KEY_KP_Home:
+            result = LV_KEY_HOME;
+            break
```
Comment (PGNetHun) [inline src/drivers/libinput/lv_xkb.c]: need `;` at the end

File: src/drivers/sdl/lv_sdl_keyboard.c [hunk 3]
```diff
@@ -204,6 +204,9 @@ static uint32_t keycode_to_ctrl_key(SDL_Keycode sdl_key)
         case SDLK_PAGEUP:
             return LV_KEY_PREV;
 
+        case SDLK_HOME:
```
Comment (PGNetHun) [inline src/drivers/sdl/lv_sdl_keyboard.c:207]: Please add `SDLK_END`, too

**Review by PGNetHun — CHANGES_REQUESTED** [review]

File: src/drivers/libinput/lv_xkb.c (hunk 2 above)
Comment (ThomasDevoogdt) [reply src/drivers/libinput/lv_xkb.c]: Oops, I guess it was better to start the weekend. ;)

**Review by PGNetHun — APPROVED** [review]

**Review by XuNeo — APPROVED** [review]

**Pull request merged**